tests/unit/test_parser.py

```diff
@@ -64,6 +64,36 @@ class RootTestParser(parser.RootParser):
     start = StartParser
 
 
+class RootTestParserWithExtra(parser.RootParser):
```

Comment:
It's not really clear what's going on here with all of these nested inner classes. Can you please provide a brief docstring or comments (or both?) on how this works?

Reply:
fixed